dll/shellext/deskadp/deskadp.c

```diff
@@ -308,28 +309,40 @@ InitDisplayAdapterDialog(PDESKDISPLAYADAPTER This)
         LocalFree((HLOCAL)lpAdapterName);
     }
 
+    LoadString(hInstance,
+               IDS_NOTAVAIL,
+               lpNA,
+               _countof(lpNA));
+
     if (This->DeskExtInterface != NULL)
     {
         SetDlgItemTextW(This->hwndDlg,
                         IDC_CHIPTYPE,
-                        This->DeskExtInterface->ChipType);
+                        wcslen(This->DeskExtInterface->ChipType) ? This->DeskExtInterface->ChipType : lpNA);
         SetDlgItemTextW(This->hwndDlg,
                         IDC_DACTYPE,
-                        This->DeskExtInterface->DacType);
+                        wcslen(This->DeskExtInterface->DacType) ? This->DeskExtInterface->ChipType : lpNA);
         SetDlgItemTextW(This->hwndDlg,
                         IDC_MEMORYSIZE,
-                        This->DeskExtInterface->MemorySize);
+                        wcslen(This->DeskExtInterface->MemorySize) ? This->DeskExtInterface->ChipType : lpNA);
         SetDlgItemTextW(This->hwndDlg,
                         IDC_ADAPTERSTRING,
-                        This->DeskExtInterface->AdapterString);
+                        wcslen(This->DeskExtInterface->AdapterString) ? This->DeskExtInterface->ChipType : lpNA);
         SetDlgItemTextW(This->hwndDlg,
                         IDC_BIOSINFORMATION,
-                        This->DeskExtInterface->BiosString);
+                        wcslen(This->DeskExtInterface->BiosString) ? This->DeskExtInterface->ChipType : lpNA);
 
         This->lpDevModeOnInit = This->DeskExtInterface->GetCurrentMode(This->DeskExtInterface->Context);
     }
     else
+    {
         This->lpDevModeOnInit = NULL;
+        SetDlgItemTextW(This->hwndDlg, IDC_CHIPTYPE, lpNA);
+        SetDlgItemTextW(This->hwndDlg, IDC_DACTYPE, lpNA);
+        SetDlgItemTextW(This->hwndDlg, IDC_ADAPTERSTRING, lpNA);
+        SetDlgItemTextW(This->hwndDlg, IDC_MEMORYSIZE, lpNA);
```

Comment:
```suggestion
        SetDlgItemTextW(This->hwndDlg, IDC_MEMORYSIZE, lpNA);
        SetDlgItemTextW(This->hwndDlg, IDC_ADAPTERSTRING, lpNA);
```
(Same order as the SetDlgItemTextW calls above + order of controls in the dialog.)